database/instance_db.go

```diff
@@ -120,7 +126,31 @@ func (d *LevelDBInstanceDB) Close() error {
 	return d.db.Close()
 }
 
-// Delete deletes service from database.
+// Delete deletes an instance from database.
 func (d *LevelDBInstanceDB) Delete(hash hash.Hash) error {
 	return d.db.Delete(hash, nil)
 }
+
+// ErrInstanceNotFound is an not found error.
+type ErrInstanceNotFound struct {
+	hash hash.Hash
+}
+
+func (e *ErrInstanceNotFound) Error() string {
+	return fmt.Sprintf("database: instance %q not found", e.hash)
+}
+
+// InstanceDecodeError represents a resource impossible to decode.
+type InstanceDecodeError struct {
```

Comment:
Do not create dedicated errors unless you want to use them.

Look comment below on decode error as well.